packages/components/src/components/multi-select/multi-select/multi-select.tsx

```diff
@@ -0,0 +1,334 @@
+                {isRequiredAndParentNotRequired(this.host, this.nativeSelect) && <Required />}
+              </span>
+            )}
+            {/* TODO: Description should be separated from the label (affects all form components) */}
+            {hasDescription(this.host, this.description) && (
+              <span class="label__text" id="description" onClick={() => this.inputElement.focus()} aria-hidden="true">
+                {this.description || <slot name="description" />}
+              </span>
+            )}
+          </label>
+          <div class={{ 'input-container': true, disabled: this.disabled }} ref={(el) => (this.inputContainer = el)}>
+            <input
+              id="comboxbox"
+              role="combobox"
+              placeholder={this.selectedString || null}
+              autoComplete="off"
+              onInput={this.onFilterChange}
+              onClick={this.onInputClick}
+              disabled={this.disabled}
+              required={this.required}
+              onKeyDown={this.onComboboxKeyDown}
+              ref={(el) => (this.inputElement = el)}
+              {...getFilterInputAriaAttributes(this.isOpen, this.required, 'label', 'description', dropdownId)}
+            />
+            <PrefixedTagNames.pButtonPure
+              class="icon reset-icon"
+              icon="close"
+              hideLabel="true"
+              theme={this.theme}
+              color={this.disabled ? 'state-disabled' : 'primary'}
+              onClick={this.onResetClick}
+            >
+              Reset selection
+            </PrefixedTagNames.pButtonPure>
+            <PrefixedTagNames.pIcon
+              class={{ icon: true, ['toggle-icon']: true, ['toggle-icon--open']: this.isOpen }}
+              name="arrow-head-down"
+              theme={this.theme}
+              color={this.disabled ? 'state-disabled' : 'primary'}
+              onClick={this.onIconClick}
+              aria-hidden="true"
+            />
+          </div>
+          <ul
+            id={dropdownId}
+            {...getListAriaAttributes(this.label, this.required, true, this.isOpen, true)}
+            ref={(el) => (this.listElement = el)}
+          >
+            {!hasFilterOptionResults(this.multiSelectOptions) && (
+              <li class="no-results" aria-live="polite" role="status">
+                <span aria-hidden="true">---</span>
+                <span class="no-results__sr">No results found</span>
+              </li>
+            )}
+            <slot onSlotchange={() => this.updateOptions()} />
+          </ul>
+        </div>
+        {this.isWithinForm && <slot name="select"></slot>}
+        {hasMessage(this.host, this.message, this.state) && (
+          <StateMessage state={this.state} message={this.message} theme={this.theme} host={this.host} />
+        )}
+        <span class="sr-text" role="status" aria-live="assertive" aria-relevant="additions text">
+          {getHighlightedOption(this.multiSelectOptions) &&
```

Comment:
Maybe create a utility function for this or at least only call the same function once instead of 4 times.